File: docs/resources/source-code/ISWC-2/test/Portal.Tests/cypress/models/submission.ts
import { randomString } from 'cypress/utils/utils';

export class InterestedParty {
   name: string;
   nameNumber: number;
   baseNumber: string;
   role: string;

   constructor(name: string, nameNumber: number, baseNumber: string, role: string) {
      this.name = name;
      this.nameNumber = nameNumber;
      this.baseNumber = baseNumber;
      this.role = role;
   }
}

export class Performer {
   firstName: string = randomString(8);
   lastName: string = randomString(8);
}

export class Submission {
   agency: string;
   sourcedb: number;
   workcode: string;
   preferredIswc: string;
   category: string;
   originalTitle: string;
   interestedParties: InterestedParty[];
   performers: Performer[];
   constructor(fixture: any) {
      this.agency = fixture.agencyCode;
      this.sourcedb = fixture.sourceDb;
      this.interestedParties = [];
      fixture.ips.forEach((ip) => {
         this.interestedParties.push(new InterestedParty(ip.name, ip.nameNumber, ip.baseNumber, ip.role));
      });
      this.category = 'DOM';
      this.originalTitle = `UI TEST ${randomString(20)} ${randomString(20)}`;
      this.workcode = randomString(17);
      this.performers = [new Performer()];
      this.preferredIswc = '';
   }
}

File: docs/resources/source-code/ISWC-2/test/Performance.Tests/data/submission.py
from utils.utils import get_new_workcode, get_new_title


def new_iswc_submission():
    return {
        "agency": "003",
        "sourcedb": 300,
        "workcode": get_new_workcode(),
        "category": "DOM",
        "disambiguation": False,
        "interestedParties": [
            {
                "BaseNumber": "I-000240590-3",
                "LastName": "PAPAVRAMIDIS",
                "NameNumber": 23429210,
                "Role": 7
            },
            {
                "BaseNumber": "I-000122272-4",
                "LastName": "KANAKIS",
                "NameNumber": 255096361,
                "Role": 7
            },
            {
                "BaseNumber": "I-000272973-7",
                "LastName": "MALIARAS",
                "NameNumber": 19265386,
                "Role": 7
            },
            {
                "BaseNumber": "I-000118019-2",
                "LastName": "VOSTANJOGLOU",
                "NameNumber": 255128672,
                "Role": 7
            },
            {
                "BaseNumber": "I-000203756-9",
                "LastName": "KONTAKIS",
                "NameNumber": 272889810,
                "Role": 7
            },
            {
                "BaseNumber": "I-000203750-3",
                "LastName": "STATHOULIS",
                "NameNumber": 226045004,
                "Role": 7
            }
        ],
        "originalTitle": get_new_title()
    }
